src/linagora.esn.unifiedinbox/app/components/sidebar/mailbox-display/mailbox-display.controller.js

```diff
@@ -0,0 +1,57 @@
+const _ = require('lodash');
+
+'use strict';
+
+angular.module('linagora.esn.unifiedinbox')
+  .controller('mailboxDisplayController', mailboxDisplayController);
+
+function mailboxDisplayController(
+  $rootScope,
+  inboxCustomRoleMailboxService,
+  inboxMailboxesService,
+  inboxJmapItemService,
+  featureFlags,
+  MAILBOX_ROLE_ICONS_MAPPING,
+  INBOX_EVENTS
+) {
+  var self = this;
+
+  self.$onInit = $onInit;
+  self.toggleMenuItem = toggleMenuItem;
+  self.getMailboxIcon = getMailboxIcon;
+  self.onDrop = onDrop;
+  self.isDropZone = isDropZone;
+
+  /////
+
+  function $onInit() {
+    self.mailboxIcons = getMailboxIcon();
+
+    self.featureFlagInboxExperimentTreeFolders = featureFlags.isOn('inbox.experiment.tree-folders');
+  }
+
+  function toggleMenuItem() {
+    this.toggle({ scope: self });
+  }
+
+  $rootScope.$on(INBOX_EVENTS.BADGE_LOADING_ACTIVATED, function(evt, data) {
```

Comment:
there is a memory leak here. The listener that you setup is not unregistered on component destroy. Can you:

- move this $rootScope.$on in the $onInit function ?
- add an $onDestroy function to unregister the listener ?